### PR TITLE
Add section in upgrading guide about lack of compatibility for 3.6.

### PR DESCRIPTION
**Pull request by pasinskim**
3.6 will be no longer compatible with 3.9 and higher.
Some typos fixed.

### CODE REVIEW

**Comment by pasinskim** [comment]
@kacf @nickanderson can you check the wording?

File: guide/installation-and-configuration/upgrading.markdown [hunk 1]
```diff
@@ -9,10 +9,15 @@ This guide documents our recommendation on how to upgrade an existing
 installation of CFEngine Community 3.7 and CFEngine Enterprise 3.7 to
```
Comment (nickanderson) [inline guide/installation-and-configuration/upgrading.markdown]: and 3.8?

File: guide/installation-and-configuration/upgrading.markdown [hunk 2]
```diff
@@ -65,10 +70,10 @@ versions) below.
 2. Save the list of hosts currently connecting to the Policy Server.
    * `cf-key -s > /root/3.7/hosts`
 
-3. Prepare masterfiles folowing instrusctions in Prepare masterfiles for
+3. Prepare masterfiles following instructions in Prepare masterfiles for
```
Comment (nickanderson) [inline guide/installation-and-configuration/upgrading.markdown]: Prepare masterfiles following instructions in *the* Prepare masterfiles for

File: guide/installation-and-configuration/upgrading.markdown [hunk 3]
```diff
@@ -111,6 +116,19 @@ versions) below.
 5. Use `cf-promises` to verify that the policy runs with you previous version of CFEngine (e.g. 3.7), by running the same commands as above on a node with that CFEngine version.
 6. The merged masterfiles should now be based on the {{site.cfengine.branch}} framework, include your policies and work on both the version you are upgrading from and with {{site.cfengine.branch}}.
 
+## Upgrade from 3.6
+
+As 3.6 policy is not compatible with {{site.cfengine.branch}} some additional steps must be performed 
+to fulfill the upgrade procedure.
+
+1. From the version 3.9 legacy package promise is deprecated and new package promise is used by default.
+   There are ```body common control``` ```package_inventory``` and ```package_module``` attributes which are
```
Comment (nickanderson) [inline guide/installation-and-configuration/upgrading.markdown]: single **`** should suffice.

Perhaps we should like to the legacy package promise implementation here.

Beginning with version 3.9 the [Masterfiles Policy Framework][The Policy Framework] defaults to the new packages promise implementation for inventory of [packages installed][`packagesmatching()`] and [packages updates][`packageupdatesmatching()`]. See [package_inventory] in `body common control` for details on modifying the default sources for package inventory.

File: guide/installation-and-configuration/upgrading.markdown [hunk 4]
```diff
@@ -9,10 +9,15 @@ This guide documents our recommendation on how to upgrade an existing
...
+We recommend upgrading the Masterfiles Policy Framework first so that you can
 verify that the new policy will work with your old clients across the
 infrastructure. Once the latest policy has been deployed successfully we
-reccomend upgrading the Policy Server and finally the remote agents.
+recommend upgrading the Policy Server and finally the remote agents.
+
+Upgrading to {{site.cfengine.branch}} from versions older than 3.7 is more
+complicated as some functionalities introduced in {{site.cfengine.branch}}
+are not compatible with versions 3.6 and earlier. 
+For more information about upgrading from 3.6 see [Upgrade from 3.6](#upgrade-from-36).
```
Comment (kacf) [inline guide/installation-and-configuration/upgrading.markdown]: as some *functionality* introduced in {{site.cfengine.branch}}
*is* not compatible

**Comment by kacf** [comment]
Looks good other than small nitpicks.

File: guide/installation-and-configuration/upgrading.markdown (hunk 1 above)
Comment (pasinskim) [inline guide/installation-and-configuration/upgrading.markdown]: Yup, makes sense to add 3.8 here as well.

File: guide/installation-and-configuration/upgrading.markdown (hunk 2 above)
Comment (pasinskim) [inline guide/installation-and-configuration/upgrading.markdown]: Fixing!

**Comment by pasinskim** [comment]
All issues fixed.

**Pull request merged**